plugins/aws/sts_provisioner.go

```diff
@@ -77,3 +95,263 @@ func (p STSProvisioner) Deprovision(ctx context.Context, in sdk.DeprovisionInput
+	}
+
+	if awsConfig.NonChainedGetSessionTokenDuration == 0 {
+		awsConfig.NonChainedGetSessionTokenDuration = 15 * time.Minute // default to minimum duration of 15 minutes
+	}
+
+	return &mfaSessionTokenProvider{
+		SessionTokenProvider: confighelpers.SessionTokenProvider{
+			StsClient: getSTSClient(awsConfig.Region, m.NewAccessKeysProvider()),
+			Duration:  awsConfig.NonChainedGetSessionTokenDuration,
+			Mfa:       confighelpers.NewMfa(awsConfig),
+		},
+		stsCacheWriter: NewSTSCacheWriter(mfaCacheKey, m.OutCache),
+	}
+}
+
+func (m CacheProviderFactory) NewAccessKeysProvider() aws.CredentialsProvider {
+	return accessKeysProvider{itemFields: m.ItemFields}
+}
+
+// getAWSAuthConfigurationForProfile loads specified configurations from both config file and environment
+func getAWSAuthConfigurationForProfile(profile string) (*confighelpers.Config, error) {
+	// Read config file from the location set in AWS_CONFIG_FILE env var or from  ~/.aws/config
+	configFile, err := confighelpers.LoadConfigFromEnv()
+	if err != nil {
+		return nil, err
+	}
+
+	configLoader := confighelpers.ConfigLoader{
+		File:          configFile,
+		ActiveProfile: profile,
+	}
+
+	// loads configuration from both environment and config file
+	configuration, err := configLoader.LoadFromProfile(profile)
+	if err != nil {
+		return nil, err
+	}
+
+	return configuration, nil
+}
+
+// resolveLocalAnd1PasswordConfigurations goes over configurations present in both local settings and 1Password and resolves conflicts using the following rules:
+// - if a certain configuration is present only in 1Password, use that one.
+// - if a certain configuration is present only in local configs, use that one.
+// - if a certain configuration is present in both places, validate that its value is consistent between the two and use it, otherwise return an error
+func resolveLocalAnd1PasswordConfigurations(itemFields map[sdk.FieldName]string, awsConfig *confighelpers.Config) error {
+	mfaSerial, hasMFASerial := itemFields[fieldname.MFASerial]
+	totp, hasOTP := itemFields[fieldname.OneTimePassword]
+	region, hasRegularRegion := itemFields[fieldname.Region]
+	defaultRegion, hasDefaultRegion := itemFields[fieldname.DefaultRegion]
+	if hasDefaultRegion {
+		region = defaultRegion
+	}
+
+	hasRegion := hasDefaultRegion || hasRegularRegion
+
+	// only 1Password OTPs are supported
+	if awsConfig.MfaToken != "" || awsConfig.MfaProcess != "" || awsConfig.MfaPromptMethod != "" {
+		return fmt.Errorf("only 1Password-backed OTP authentication is supported by the MFA worklfow of the AWS shell plugin")
+	}
+	// make sure 1Password OTP is used
+	if hasOTP {
+		awsConfig.MfaToken = totp
+	}
+
+	if hasMFASerial && awsConfig.HasMfaSerial() && awsConfig.MfaSerial != mfaSerial {
+		return fmt.Errorf("your local AWS configuration (config file or environment variable) has a different MFA serial than the one specified in 1Password")
+	} else if !awsConfig.HasMfaSerial() {
```

Comment:
Yes, that is true, but the reason I expressed this as an `else if` is because if I wanted to achieve the same logic without the else I would have had to write:

```
if !awsConfig.HasMfaSerial() && hasMfaSerial {
}
```
which was a bit longer. 
Given this is more a preference detail, I'll choose to not commit a change because I'll be loosing the acquired approvals.